placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+                targetRoundtablePosition.y.toFixed(FIXED_DECIMAL_PLACES);
+                if ((roundtableVelocity < 0 && roundtableDistanceToTarget <= 0) || 
+                    (roundtableVelocity > 0 && roundtableDistanceToTarget >= 0)) {
+                    roundtableInPlace = true;
+                }
+                if (roundtableInPlace) {
+                    roundtableVelocity = { x: 0, y: 0, z: 0 };
+                    Entities.editEntity(roundtable, {
+                        velocity: { x: 0, y: 0, z: 0 },
+                        position: targetRoundtablePosition
+                    });
+                }
+            }
+
+            if (courtyardInPlace && roundtableInPlace) {
+                _this.playSound(DONE_SOUND, DONE_VOLUME, courtyardCurrentPosition, false, false);
+
+                Entities.editEntity(buttonID, {
+                    backgroundColor: BUTTON_ACTIVE_COLOR,
+                    textColor: BUTTON_INACTIVE_COLOR
+                });
+                
+                if (checkInterval) {
+                    Script.clearInterval(checkInterval);
+                    checkInterval = null;
+                }
+            }
+        },
+
+        /* Make sure this entity is in a reasonable place to begin movement. If it is too far away slam it into position 
+        beneath the stage. If it is already very close to its target position, slam it into exact position and 
+        always ensure it is aligned correctly on the x and z axes. Use its distance to its target positon to determine 
+        the velocity it needs to get there. */ 
+        setUpMovement: function(entityID, targetPosition) {
+            var currentPosition = Entities.getEntityProperties(entityID, 'position').position;
+    
+            // This part will ensure that the observer sees movement immediately
+            if (entityID === roundtable) {
+                var courtyardProperties = Entities.getEntityProperties(courtyard, ['dimensions', 'position']);
+                var courtyardHeight = courtyardProperties.dimensions.y;
+                
+                // if the top of the table is below the top of the courtyard, reposition it to be just the top of 
+                // the courtyard. Using the top of the courtyard gives a little leeway to ensure the table ends up below it
+                if (currentPosition.y + tableHeight * HALF <= courtyardProperties.position.y + courtyardHeight * HALF) {
+                    currentPosition = {
+                        x: currentPosition.x,
+                        y: courtyardProperties.position.y - tableHeight * HALF,
+                        z: currentPosition.z
+                    };
+                    Entities.editEntity(roundtable, { position: currentPosition });
+                }
+            }
+            
+            var distanceToTarget = currentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
```

Comment:
- Why are you using `toFixed` here?
- Why not use `Vec3.distance()` here?
- This isn't a distance, since it could be negative. Did you intend for this to be a distance? If so, you'll need to wrap this in `Math.abs()` (but only if you're not going to use `Vec3.distance()`)

Reply:
https://github.com/highfidelity/hifi-content/pull/462#discussion_r325268910